Inspector: Add new AdvancedSlider control

EchterAgo opened the pull request:
Replace the old RangeEditor and TextBox numeric editors with the new
advanced slider control. The user can use a combination of ALT, CTRL and
SHIFT keys to adjust the rate of change. The user can also double click
the value to edit the value numerically.

Signed-off-by: Axel Gembe <axel@gembe.net>

EchterAgo commented:
This is a large one. Sadly it can not be used with nullable values.

EchterAgo commented on src/Gemini.Modules.Inspector/Conventions/AdvancedSliderRangePropertyEditorBuilder.cs:
```diff
@@ -0,0 +1,53 @@
+            if (propertyDescriptor.PropertyType == typeof(double))
+            {
+                return new AdvancedSliderEditorViewModel<double>((double) rangeAttribute.Minimum, (double) rangeAttribute.Maximum) {
+                    Speed = 0.1
```
These should maybe come from an attribute as well.

EchterAgo commented:
Rebased on top of #192

EchterAgo commented:
Updated patch to make it easier to set the cursor inside the text field and to make the text box use the key down event for ending editing so that other controls do not handle the key down event first.

EchterAgo commented:
Updated to support keyboard focus

tgjones commented:
This looks quite nice. I'm concerned about the overlap with `NumericTextBox` though (as seen in the MonoGame demo, in the `Vector3` property editor). Could we extend `NumericTextBox` instead of introducting a new control?

In any case, I'll wait till #192 is merged, so that the diff here is easier to review.

EchterAgo commented:
Rebased

EchterAgo commented:
I'll do that, but I'd like to look into addressing your concerns regarding `NumericTextBox` overlap first. I think you should take a look at merging #298, it is a simple bugfix.
